packages/nft-erc1155-provider/lib/NftErc1155Provider.ts

```diff
@@ -0,0 +1,87 @@
+import { addressToString } from '@liquality/utils'
+import { ensure0x, normalizeTransactionObject } from '@liquality/ethereum-utils'
+import { Address } from '@liquality/types'
+import { StandardError } from '@liquality/errors'
+import { NftBaseProvider } from '@liquality/nft-base-provider'
+
+import { isArray } from 'lodash'
+import { Signer } from '@ethersproject/abstract-signer'
+
+import NftErc1155_ABI from './NftErc1155_ABI.json'
+const erc1155InterfaceID = '0xd9b67a26'
+
+export default class NftErc1155Provider extends NftBaseProvider {
+  constructor(signer: Signer) {
+    super(signer, NftErc1155_ABI as any, erc1155InterfaceID)
+  }
+
+  async balance(contract: Address | string, tokenIDs: number | number[]) {
+    if (!tokenIDs) {
+      return 0
+    }
+
+    await super.balance(contract, tokenIDs)
```

Comment:
From here it's not clear what the super function is doing. If all the super functions are just attaching the contract and caching it, then I think its better to call it something else.

like `this.setContract(contract)`

as a result I think the super methods are unnecessary